### DISCUSSION TITLE
How can I deploy pocketbase in AWS EC2?

### DISCUSSION THREAD

**Discussion by fjaviermoradiaz** [pocketbase/pocketbase#297] [Q&A] [answered]
I'm trying to deploy pocketbase in AWS EC2 service (I'v tested with ec2 ubuntu and amazon linux 2)

command:

`sudo ./pocketbase serve --http="ec2-XX-XXX-XX-XXX.region-1.compute.amazonaws.com:80" --https="ec2-XX-XXX-XX-XXX.region-1.compute.amazonaws.com:443"`

log:

  ```
Server started at: https://ec2-XX-XXX-XX-XXX.region-1.compute.amazonaws.com:443
  - REST API: https://ec2-XX-XXX-XX-XXX.region-1.compute.amazonaws.com:443/api/
  - Admin UI: https://ec2-XX-XXX-XX-XXX.region-1.compute.amazonaws.com:443/_/
```

But when I try to access tohttps://ec2-XX-XXX-XX-XXX.region-1.compute.amazonaws.com:443/_/ I get this message:

```
2022/08/12 06:29:01 http: TLS handshake error from 91.116.2.152:53669: 400 urn:ietf:params:acme:error:rejectedIdentifier: Error creating new order :: Cannot issue for "ec2-XX-XXX-XX-XXX.region-1.compute.amazonaws.com": The ACME server refuses to issue a certificate for this domain name, because it is forbidden by policy
2022/08/12 06:29:01 http: TLS handshake error from 91.116.2.152:54230: acme/autocert: missing certificate
2022/08/12 06:29:01 http: TLS handshake error from 91.116.2.152:54121: acme/autocert: missing certificate
2022/08/12 06:29:01 http: TLS handshake error from 91.116.2.152:54110: acme/autocert: missing certificate
```

Has anyone been able to deploy it?

**Comment by ganigeorgiev** [comment, marked as answer]
This is a limitation of Let's encrypt - https://community.letsencrypt.org/t/policy-forbids-issuing-for-name-on-amazon-ec2-domain/12692/3.

Let's encrypt doesn't allow issuing certificates on AWS domains because the domain names are transient and are subject to change.

**Reply by fjaviermoradiaz** [reply]
do you know if the same thing happens with google cloud?

**Reply by ganigeorgiev** [reply]
Sorry I'm not sure, but if their generated domain names are ephemeral like the AWS, then I guess  the same policy will be applied.

You can still use your AWS instance, just bring your own domain - the Let's encrypt moderator in the above link suggested http://www.dot.tk/ as a free alternative.

**Reply by fjaviermoradiaz** [reply]
Thanks [ganigeorgiev](https://github.com/ganigeorgiev) !!!

**Reply by rodydavis** [reply]
I ran into the same issue with GCP on compute engine

**Reply by helblinglilly** [reply]
You can overcomplicate it, I'm working on a terraform template and it will be ready to publish soon (just need to get Email going, will update here when ready)

If you get a domain into Route 53 (or create a subdomain and point an NS record at AWS) you can get AWS to issue SSL certificates to that domain.
You'll then create an A Alias record to point at an Application Load Balancer. That ALB will need to listen to traffic on 443 and route it to a target group on the port on which you run Pocketbase in http only mode (let's say 8090). That target group must point at the EC2 instance running Pocketbase.

So pocketbase will just run in http mode and you're effectively using the ALB as a tool to map ports.

I'm not saying this is a well architected solution, but it seems to work for me at least.

**Reply by helblinglilly** [reply]
Couldn't get SES working, but as promised: https://github.com/helblingjoel/aws-pocketbase

See the ![Architecture Diagram](https://github.com/helblingjoel/aws-pocketbase/blob/main/architecture.png). I'm not saying it's great or follows best practices, but it seems to work.
